### PR TITLE
Add ElastiCache template, update README

### PR DESCRIPTION
**Pull request by john**
Added the elasticache template. Lets you select the engine type, instance type for nodes, and number of nodes. Updated the README to match

A future update will let you opt for AZ mode, and will add launch stack links.

### CODE REVIEW

File: templates/elasticache.cfn.yml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+          FromPort: !If [ IsRedis, 6379, 11211]
+          ToPort: !If [ IsRedis, 6379, 11211]
+      
+  SubnetGroup: 
+    Type: AWS::ElastiCache::SubnetGroup
+    Properties: 
+      Description: Cache Subnet Group
+      SubnetIds:
+        - Fn::ImportValue: !Sub "${NetworkStackName}-PublicSubnet1ID"
+        - Fn::ImportValue: !Sub "${NetworkStackName}-PrivateSubnet2ID"
```
Comment (koroshan) [inline templates/elasticache.cfn.yml]: PrivateSubnet?

Comment (koroshan) [reply templates/elasticache.cfn.yml]: Node count must be between 1 and 10?

File: templates/elasticache.cfn.yml [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+          
+  ElasticacheCluster:
+    Type: AWS::ElastiCache::CacheCluster
+    Properties:
+      AutoMinorVersionUpgrade: true
```
Comment (koroshan) [inline templates/elasticache.cfn.yml]: Should we let users select this? Not sure!

File: templates/elasticache.cfn.yml [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    AllowedValues:
+      - redis
+      - memcached
+      
+  CacheNodeCount:
+    Description: Number of nodes in the cluster
+    Type: Number
+    MinValue: 1
+    MaxValue: 10
+    ConstraintDescription: Node count must be between 1 and 100
```
Comment (koroshan) [inline templates/elasticache.cfn.yml]: Node count must be between 1 and 10?

Comment (koroshan) [inline templates/elasticache.cfn.yml]: Node count must be between 1 and 10?

File: templates/elasticache.cfn.yml (hunk 2 above)
Comment (koroshan) [reply templates/elasticache.cfn.yml]: PrivateSubnet1ID?

File: templates/elasticache.cfn.yml [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+          IpProtocol: tcp
+          FromPort: !If [ IsRedis, 6379, 11211]
+          ToPort: !If [ IsRedis, 6379, 11211]
+      
+  SubnetGroup: 
+    Type: AWS::ElastiCache::SubnetGroup
+    Properties: 
+      Description: Cache Subnet Group
+      SubnetIds:
+        - Fn::ImportValue: !Sub "${NetworkStackName}-PublicSubnet1ID"
```
Comment (koroshan) [inline templates/elasticache.cfn.yml]: PrivateSubnet1ID?

File: templates/elasticache.cfn.yml (hunk 3 above)
Comment (koroshan) [reply templates/elasticache.cfn.yml]: The max for Redis is 15 and Memcached is 20. Maybe we can keep it max to 15 for now? or else we will need to implement some logic here.

File: templates/elasticache.cfn.yml (hunk 2 above)
Comment (koroshan) [reply templates/elasticache.cfn.yml]: Maybe we should add a parameter for this. Thoughts?

Comment (john) [reply templates/elasticache.cfn.yml]: Updated in next commit

File: templates/elasticache.cfn.yml (hunk 3 above)
Comment (john) [reply templates/elasticache.cfn.yml]: In next commit

File: templates/elasticache.cfn.yml (hunk 4 above)
Comment (john) [reply templates/elasticache.cfn.yml]: fixed

File: templates/elasticache.cfn.yml [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    Type: Number
+    MinValue: 1
+    MaxValue: 15
+    ConstraintDescription: Node count must be between 1 and 15
+    Default: 1
+    
+  AutoMinorVersionUpgrade:
+    Description: Whether or not minor version upgrades to the cache engine should be applied automatically during the maintenance window.
+    Type: String
+    Default: "true"
```
[the remark this answers is not in the document]
Comment (john) [reply templates/elasticache.cfn.yml]: 👍

File: templates/elasticache.cfn.yml [hunk 6]
```diff
@@ -0,0 +1,159 @@
+
+# support replica count?
```
[the remark this answers is not in the document]
Comment (john) [reply templates/elasticache.cfn.yml]: 👍

File: templates/elasticache.cfn.yml [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+  IsRedis: !Equals [ !Ref CacheEngine, redis]
+  
+Resources:
+
+  SecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Elasticache Security Group
+      VpcId:
+        Fn::ImportValue: !Sub "${NetworkStackName}-VpcID"
```
[the remark this answers is not in the document]
Comment (john) [reply templates/elasticache.cfn.yml]: 👍

File: templates/elasticache.cfn.yml [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+          IpProtocol: tcp
+          FromPort: !If [ IsRedis, 6379, 11211]
+          ToPort: !If [ IsRedis, 6379, 11211]
+      
+  SubnetGroup: 
+    Type: AWS::ElastiCache::SubnetGroup
+    Properties: 
+      Description: Cache Subnet Group
+      SubnetIds:
+        - Fn::ImportValue: !Sub "${NetworkStackName}-PrivateSubnet1ID"
```
[the remark this answers is not in the document]
Comment (john) [reply templates/elasticache.cfn.yml]: 👍

File: templates/elasticache.cfn.yml [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+      AutoMinorVersionUpgrade: !Ref AutoMinorVersionUpgrade
+      Engine: !Ref CacheEngine
+      CacheNodeType: !Ref CacheNodeType
+      NumCacheNodes: !If [ IsRedis, 1, !Ref CacheNodeCount]
+      CacheSubnetGroupName: !Ref SubnetGroup
+      VpcSecurityGroupIds:
+        - !GetAtt SecurityGroup.GroupId
+      Tags:
+        - Key: Name
+          Value: !Ref "AWS::StackName"
```
[the remark this answers is not in the document]
Comment (john) [reply templates/elasticache.cfn.yml]: 👍

File: templates/elasticache.cfn.yml [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+    Description: Elasticache Cluster Arn
+    Value: !Sub arn:aws:elasticache:${AWS::Region}:${AWS::AccountId}:cluster/${ElasticacheCluster}
+    Export:
+      Name: !Sub ${AWS::StackName}-ElastiCacheClusterArn
+      
+  ElastiCacheClusterId:
+    Description: Elasticache Cluster ID
+    Value: !Ref ElasticacheCluster
+    Export:
+      Name: !Sub "${AWS::StackName}-ElastiCacheClusterID"
```
[the remark this answers is not in the document]
Comment (john) [reply templates/elasticache.cfn.yml]: 👍

File: templates/elasticache.cfn.yml [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+      
+  ElastiCacheEngine:
+    Description: Elasticache engine
+    Value: !Ref CacheEngine
+    Export:
+      Name: !Sub "${AWS::StackName}-ElastiCacheEngine"
```
[the remark this answers is not in the document]
Comment (john) [reply templates/elasticache.cfn.yml]: 👍

File: templates/elasticache.cfn.yml [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+      
+  SubnetGroup: 
+    Type: AWS::ElastiCache::SubnetGroup
+    Properties: 
+      Description: Cache Subnet Group
+      SubnetIds:
+        - Fn::ImportValue: !Sub "${NetworkStackName}-PrivateSubnet1ID"
+        - Fn::ImportValue: !Sub "${NetworkStackName}-PrivateSubnet2ID"
+          
+  ElasticacheCluster:
```
[the remark this answers is not in the document]
Comment (john) [reply templates/elasticache.cfn.yml]: Replaced with correct capitalization across the board

File: templates/elasticache.cfn.yml [hunk 13]
```diff
@@ -0,0 +1,159 @@
...
+      Name: !Sub "${AWS::StackName}-ElastiCacheEngine"
+      
+  ElastiCacheAddress:
+    Description: Elasticache endpoint address
+    Value: !If [ IsRedis, !GetAtt ElasticacheCluster.RedisEndpoint.Address, !GetAtt ElasticacheCluster.ConfigurationEndpoint.Address]
+    Export:
+      Name: !Sub "${AWS::StackName}-ElastiCacheAddress"
+      
+  ElastiCachePort:
+    Description: Elasticache port
```
[the remark this answers is not in the document]
Comment (john) [reply templates/elasticache.cfn.yml]: I get the intent and agree, but not sure where you suggest using GetAtt. Would it accomplish the goal to just remove ElastiCachePort as an output?

**Pull request merged**